tests/integration/test_paymnet_methods.py

```diff
@@ -47,3 +49,77 @@ def test_list(self, api):
         self.assert_list_response_contains_exactly_by_id(
             payment_methods_after_last_id, [payment_method2, payment_method1]
         )
+
+    def test_will_not_create_duplicate_if_same_idempotency_key_is_used(self, api):
+        # given
+        request_options = RequestOptions()
+        request_options.set_idempotency_key(random_string())
+        payment_method_req = payment_method()
+
+        # when
+        first_call_response = api.payment_methods.create(
+            payment_method_req, request_options=request_options
+        )
+        second_call_response = api.payment_methods.create(
+            payment_method_req, request_options=request_options
+        )
+
+        # then
+        assert first_call_response == second_call_response
+
+    def test_will_create_two_instances_if_different_idempotency_keys_are_used(
+        self, api
+    ):
+        # given
+        request_options = RequestOptions()
+        request_options.set_idempotency_key(random_string())
```

Comment:
this should be doable inline with request. Python devs are not used to verbose objective API.

Something like this may be possible?
```python
customer = shift4.customers.create({
    'email': 'user@example.com',
    'description': 'User description'
}, {
    'idempotency_key': random_string(), # this or
    idempotency_key: random_string(), #  this?
})
```